### PR TITLE
Fix back stack for embedded apps

### PR DESCRIPTION
**Pull request by henrytao-me**
### WHY are these changes introduced?

Resolves part of https://github.com/Shopify/shopify-app-bridge/issues/313

### WHAT is this pull request doing?

This PR fixes Link Component to work with back stack in embedded apps. 

## Type of change

- [x] Patch: Bug (non-breaking change which fixes an issue)
- [ ] Minor: New feature (non-breaking change which adds functionality)
- [ ] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have used `yarn changeset` to create a draft changelog entry (do NOT update the `CHANGELOG.md` files manually)
- [ ] I have added/updated tests for this change
- [ ] I have documented new APIs/updated the documentation for modified APIs (for public APIs)

### CODE REVIEW

File: .changeset/curvy-apes-beam.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@shopify/shopify-app-remix": patch
+---
+
+Fix back stack for embedded apps
```
Comment (paulomarg) [inline .changeset/curvy-apes-beam.md:5]: Mind if we just add a little more colour here? This is the main way for folks to catch up on what we're doing, so I like to be very explicit in the changelog.
```suggestion
Fix a bug that prevents using the browser's back stack for embedded apps
```

File: packages/apps/shopify-app-remix/src/react/components/RemixPolarisLink.tsx
Comment (paulomarg) [inline packages/apps/shopify-app-remix/src/react/components/RemixPolarisLink.tsx:1]: Could / should we have some tests for these changes? We could create a folder around this or add them to the existing AppProvider tests (which is the only place that will use this).

File: packages/apps/shopify-app-remix/src/react/components/RemixPolarisLink.tsx [hunk 2]
```diff
@@ -9,6 +9,83 @@ import type {
...
+          } else {
+            forwardedRef.current = anchor;
+          }
+        }
+      }}
+    />
+  );
+}) as LinkLikeComponent;
+
+function parseUrl(url: string | undefined | null) {
```
Comment (paulomarg) [inline packages/apps/shopify-app-remix/src/react/components/RemixPolarisLink.tsx:62]: Total nit, but does `undefined` vs `null` matter here? It seems like we could just use `undefined` all around?

**Review by paulomarg — COMMENTED** [review]
Just some nits, but holding off on approving until we figure out whether tests make sense for this :)

File: packages/apps/shopify-app-remix/tsconfig.json [hunk 3]
```diff
@@ -3,7 +3,8 @@
   "compilerOptions": {
     "baseUrl": ".",
     "outDir": "./dist/ts",
-    "rootDir": "src"
+    "rootDir": "src",
+    "types": ["@cloudflare/workers-types", "node", "jest", "jsdom"]
```
[the remark this answers is not in the document]
Comment (henrytao-me) [reply packages/apps/shopify-app-remix/tsconfig.json:7]: Production build uses `tsconfig.build.json`, doesn't it?

[the remark this answers is not in the document]
Comment (paulomarg) [reply packages/apps/shopify-app-remix/tsconfig.json:7]: But it extends that one. We could also override the build config to set all types but jsdom, that would work :)

**Review by MitchLillie — APPROVED** [review]
🎩 via https://github.com/Shopify/app-bridge-next/pull/345

Agree on adding unit tests :)

**Comment by henrytao-me** [comment]
Close as we are working on a solution that doesn't involve update in the template

**Pull request closed without merging**